app/models/tabbed.rb

```diff
@@ -46,36 +46,61 @@ def norm(s)
       s.downcase.gsub(/\W+/, '-')
     end
 
+    def shared_prefix(titles)
+      tokens = titles.first.scan(/\S+\s+/)
+      prefixes = 0.upto(tokens.length).map { |i| tokens[0..i].join }.reverse
```

Comment:
@mccalluc may not be consequential here, but I don't think it should be `tokens.length - 1`